lib/resque/pool.rb

```diff
@@ -344,7 +344,42 @@ def quit_excess_workers_for(queues)
     end
 
     def worker_delta_for(queues)
-      config.fetch(queues, 0) - workers.fetch(queues, []).size
+      config_get_worker_count(queues).to_i - workers.fetch(queues, []).size
+    end
+
+    # Get the number of workers declared in the config file for queues
+    #
+    # @param queues [String] names of queues separated by comma if many exist
+    def config_get_worker_count(queues)
+      worker_count = nil
+      if config.has_key?(queues)
+        workers_entry = config[queues].find { |entry| entry.has_key?('workers') }
+        return worker_count unless workers_entry
+        worker_count = workers_entry['workers']
+      end
+
+      worker_count
+    end
+
+    # Check if certain queues should have their workers to fork or not
+    #
+    # NOTE: If you need to have dedicated non-forking workers for a queue then
+    # you need to correctly specify false into the `fork` key of your queue.
+    # An example can be found in the README.
+    #
+    # Default behavior is to fork in order to comply with resque since it's default
+    # behavior is also to fork unless specified otherwise (See the FORK_PER_JOB
+    # environment variable in resque).
+    #
+    # @param queues [String] names of queues separated by comma if many exist
+    def fork_enabled_for_queues?(queues)
+      queues_key = queues.is_a?(Array) ? queues.join(',') : queues
+      return unless config.has_key?(queues_key)
```

Comment:
Since this method's name ends with a question mark, we should return a boolean value.

Reply:
Fixed!